### PR TITLE
Pin 2649 [FILTER – Erogatore] BFF GET /agreements/eservices

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala [hunk 1]
```diff
@@ -490,4 +492,49 @@ final case class AgreementsApiServiceImpl(
     }
   }
 
+  override def getAgreementEServiceProducers(q: Option[String], offset: Int, limit: Int)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerCompactAgreementEServices: ToEntityMarshaller[CompactEServicesLight],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    logger.info(s"Retrieving eservices from agreement filtered by eservice name $q, offset $offset, limit $limit")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala]:
```suggestion
    logger.info(s"Retrieving producer eservices from agreement filtered by eservice name $q, offset $offset, limit $limit")
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/AgreementProcessService.scala [hunk 2]
```diff
@@ -45,4 +45,12 @@ trait AgreementProcessService {
   def removeConsumerDocument(agreementId: UUID, documentId: UUID)(implicit
     contexts: Seq[(String, String)]
   ): Future[Unit]
+
+  def getAgreementEServices(
+    eServiceName: Option[String],
+    producersIds: Seq[UUID] = Seq.empty,
+    consumersIds: Seq[UUID] = Seq.empty,
+    limit: Int,
+    offset: Int = 0
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/AgreementProcessService.scala]: let's avoid default values if not strictly required
```suggestion
    offset: Int
```

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -2832,6 +2832,133 @@ paths:
...
+                type: integer
+              description: Max allowed requests within time interval
+            'X-Rate-Limit-Remaining':
+              schema:
+                type: integer
+              description: Remaining requests within time interval
+            'X-Rate-Limit-Interval':
+              schema:
+                type: integer
+              description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: Unused status
```suggestion
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala [hunk 4]
```diff
@@ -490,4 +492,51 @@ final case class AgreementsApiServiceImpl(
...
+      }
+    }
+  }
+
+  private def validateQueryName(q: Option[String]): Future[Unit] = {
+    q match {
+      case Some(value) if value.length < 3 => Future.failed(InvalidQueryParameter(value))
+      case _                               => Future.successful(())
+    }
+  }
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala:541]: ditto

Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala]: [ditto](https://github.com/pagopa/interop-be-backend-for-frontend/pull/105#discussion_r1108706548)

**Review by TonioGela — CHANGES_REQUESTED** [review]

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**